service/matching/physical_task_queue_manager_test.go

```diff
@@ -98,10 +97,12 @@ func makeTestBlocAlloc(f func() (taskQueueState, error)) taskQueueManagerOpt {
 
 func withIDBlockAllocator(ibl idBlockAllocator) taskQueueManagerOpt {
 	return func(tqm *physicalTaskQueueManagerImpl) {
-		tqm.backlogMgr.taskWriter.idAlloc = ibl
+		blm := tqm.backlogMgr.(*backlogManagerImpl)
+		blm.taskWriter.idAlloc = ibl
 	}
 }
 
+/* TODO(pri): test is too awkward to fix and low value
```

Comment:
Let me know if you disagree.

I'd actually like to remove the whole `idBlockAllocator` interface, I think it's a confusing way to test things. If we need to inject errors to UpdateTaskQueue we should do it with our in-mem TaskManager

Reply:
👍 re `idBlockAllocator`

Reply:
I'll do it in another PR